### PR TITLE
Camera Change Listener v2.0

### LINKED ISSUE #6350: Camera start, ongoing, and end events notifications
The latest Google Maps has added camera change listeners for camera start, ongoing, and end events notifications. The interfaces are `OnCameraMoveStartedListener`, `OnCameraMoveListener`, `OnCameraMoveCanceledListener`, `OnCameraIdleListener`. This is how it's described in the [announcement](https://googlegeodevelopers.blogspot.com/2016/08/keep-users-focused-on-whats-important.html):

> As one of our top requests, developers have been asking for a better way to track camera movements and the ability to see why the camera is moving, whether caused by user gestures, built-in API animations or developer controlled movements [Issue 4636]. Our new camera change listeners support you in doing this. Your app can now receive notifications for camera start, ongoing, and end events.

I hear these are useful and we should consider supporting them as well for API parity.

/cc: @cammace @tobrun

### PR DESCRIPTION
**Pull request by tobrun**
Closes #6350.

### CODE REVIEW

**Comment by tobrun** [comment]
I took another stab at optimizing the onFling integration. This PR is now ready for review.

cc @Guardiola31337 @zugaldia

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java [hunk 1]
```diff
@@ -187,6 +212,16 @@ void setZoom(double zoom, @NonNull PointF focalPoint) {
   }
 
   void setZoom(double zoom, @NonNull PointF focalPoint, long duration) {
+    mapView.addOnMapChangedListener(new MapView.OnMapChangedListener() {
+      @Override
+      public void onMapChanged(int change) {
+        Timber.e("CHANGE : %s", change);
+        if (change == MapView.REGION_DID_CHANGE_ANIMATED) {
+          mapView.removeOnMapChangedListener(this);
```
Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java]: Wondering if it order matters 🤔 
In `setZoom` is defined `mapView.removeOnMapChangedListener` and `cameraChangeDispatcher.onCameraIdle` second but in `moveBy` is defined the other way around.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/CameraChangeDispatcher.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    this.onCameraMoveListener = onCameraMoveListener;
+  }
+
+  void setOnCameraIdleListener(OnCameraIdleListener onCameraIdleListener) {
+    this.onCameraIdleListener = onCameraIdleListener;
+  }
+
+  @Override
+  public void onCameraMoveStarted(int reason) {
+    if (!idle) {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/CameraChangeDispatcher.java]: I don't a preference for one or the other, will fix this up.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java (hunk 1 above)
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java]: doesn't change behavior, but for consistency will pick that up.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/CameraChangeDispatcher.java (hunk 2 above)
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/CameraChangeDispatcher.java]: ah, checkstyle doesn't approve this.. `[ant:checkstyle] [ERROR] /home/tvn/Mapbox/mapbox-gl-native/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/CameraChangeDispatcher.java:36: 'if' construct must use '{}'s. [NeedBraces]`

Feel free to ticket out changing this setting and doing a cleanup of the codebase.

**Pull request merged**